components/challenge_bypass_ristretto/batch_dleq_proof.cc

```diff
@@ -0,0 +1,224 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/challenge_bypass_ristretto/batch_dleq_proof.h"
+
+#include <optional>
+#include <utility>
+
+#include "base/types/expected.h"
+#include "brave/components/challenge_bypass_ristretto/blinded_token.h"
+#include "brave/components/challenge_bypass_ristretto/public_key.h"
+#include "brave/components/challenge_bypass_ristretto/signed_token.h"
+#include "brave/components/challenge_bypass_ristretto/signing_key.h"
+#include "brave/components/challenge_bypass_ristretto/token.h"
+#include "brave/components/challenge_bypass_ristretto/unblinded_token.h"
+#include "brave/third_party/rust/challenge_bypass_ristretto_cxx/src/lib.rs.h"
+
+namespace challenge_bypass_ristretto {
+
+namespace {
+
+rust::Box<cbr_cxx::BlindedTokensResult> ConvertToBlindedTokens(
+    const std::vector<BlindedToken>& blinded_tokens) {
+  std::vector<std::string> encoded_blinded_tokens;
+  encoded_blinded_tokens.reserve(blinded_tokens.size());
+  for (const auto& blinded_token : blinded_tokens) {
+    encoded_blinded_tokens.push_back(blinded_token.EncodeBase64());
+  }
+  return cbr_cxx::decode_base64_blinded_tokens(encoded_blinded_tokens);
```

Comment:
Is it necessary to round-trip through base64 here? Seems like it would be more efficient to pass the `BlindedToken` vector directly?

Reply:
This is how it is done with the original version of CXX wrapper:
https://github.com/brave-experiments/challenge-bypass-ristretto-cxx

Reply:
Do you mean in the [example code](https://github.com/brave-experiments/challenge-bypass-ristretto-cxx/blob/main/examples/main.cpp#L112-L117)? In that case it makes more since in that the client is simulating storage between creation and redemption.

Anyway, I was just pointing out it seems wasteful, or like the API wasn't designed for the use we need to make of it. It doesn't need to block landing.